File: Project/FoodProject_Frontend/ProjectFood/src/services/dashboardService.js
import api from "./api";

// Dashboard
export const getDonorDashboard = () => api.get("/dashboard/donor").then(r => r.data);
export const getAdminDashboard = () => api.get("/dashboard/admin").then(r => r.data);
export const getImpactDashboard = () => api.get("/dashboard/impact").then(r => r.data);
export const getMonthlyStatistics = () => api.get("/dashboard/statistics").then(r => r.data);

export default { getDonorDashboard, getAdminDashboard, getImpactDashboard, getMonthlyStatistics };

File: Project/Backend/src/main/java/com/food/service/DashboardService.java
package com.food.service;

import java.util.List;

import com.food.DTO.AdminDashboardDTO;
import com.food.DTO.DonorDashboardDTO;
import com.food.DTO.ImpactDashboardDTO;
import com.food.DTO.MonthlyStatisticsDTO;
import com.food.DTO.WasteDashboardDTO;

public interface DashboardService {
	// Admin dashboard
	public AdminDashboardDTO getAdminDashboard();

	// Donor dashboard
	public DonorDashboardDTO getDonorDashboard(Long userId);

	// Impact dashboard
	public ImpactDashboardDTO getImpactDashboard();

	// Monthly Statistics
	public List<MonthlyStatisticsDTO> getMonthlyStatistics();

	// waste dashboard
	public WasteDashboardDTO getWasteDashboard();
}
